polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -204,6 +204,34 @@ static void checkFunctionParent(const FunctionOpInterface F,
          "New device function must be inserted into device module");
 }
 
+/// If \p FD corresponds to a function implementing a SYCL method, registers
+/// \p Func as the function implementing the corrsponding operation.
+static void tryToRegisterSYCLMethod(const FunctionDecl &FD,
+                                    FunctionOpInterface Func) {
+  if (!mlirclang::isNamespaceSYCL(FD.getEnclosingNamespaceContext()) ||
+      !isa<CXXMethodDecl>(FD) ||
+      isa<CXXConstructorDecl, CXXDestructorDecl>(FD) ||
+      Func.getNumArguments() == 0) {
```

Comment:
The previous conditions should have taken care of filtering out unwanted functions. I'm, wondering if we need this last check, C++ member functions, constructors and destructors would all have the implicit `this` pointer argument so I think this check is redundant.

Reply:
`CXXMethodDecl` includes static functions (see [here](https://clang.llvm.org/doxygen/classclang_1_1CXXMethodDecl.html))

Reply:
I'm using `isInstance`, so we can drop that. Thanks for pointing out.